buildSrc/src/main/groovy/com/uber/okbuck/composer/groovy/GroovyLibraryRuleComposer.groovy

```diff
@@ -14,16 +14,16 @@ final class GroovyLibraryRuleComposer extends JvmBuckRuleComposer {
 
     static GroovyLibraryRule compose(GroovyLibTarget target) {
         List<String> deps = []
-        deps.addAll(external(target.main.externalDeps))
+        deps.addAll(external(target.main.externalDeps, target))
```

Comment:
these static methods should not be getting an instance of target passed into them. There is too much leakage of info here. Lets talk over slack on how we can simplify this